lib/at_notif/Kconfig

```diff
@@ -6,7 +6,7 @@
 
 menuconfig AT_NOTIF
 	bool "AT-command notification manager"
-	depends on AT_CMD
+	imply AT_CMD
```

Comment:
The answer is similar as for Even here: https://github.com/nrfconnect/sdk-nrf/pull/3698#discussion_r589247400

Similarly, I'd like to have CONFIG_SMS=y in unity test prj.conf. At the same time, I'd like to have real version of at_notif library and mock at_cmd library. Without changing this dependency, I haven't found a way to accomplish this.

Reply:
Personally, I think it's rather the test who should stretch out to fit the existing (and valid) Kconfig dependencies instead of modyfing them for purely its needs (CC @tejlmand). Why not mock the AT_NOTIF as well?

Reply:
Generally, in the tests I only want to mock as little as possible because that would reveal more issues in this early testing. We would then test more about the library and its dependencies and how those dependencies operate. If I mock at_notif and maybe at_parser, I would need to know how exactly they operate in some corner cases. 

Mocking at_notif wouldn't be a big deal as that's just two functions. But then there's the change within sms/Kconfig that Even commented earlier related to select vs imply AT_CMD.

Then one more aspect. Testability is one part of SW design so taking testing into account in the configurations is one aspect. Doesn't mean it has to be taken into account in this specific case but it wouldn't be wrong to make a compromise in the Kconfig's or code in general to enable testing easier.

Reply:
> Generally, in the tests I only want to mock as little as possible because that would reveal more issues in this early testing.

I understand, but the Kconfig should be seen from the point of our users.

If this indeed is a real depends, meaning that `AT_NOTIF` depends on `AT_CMD` then this should be reflected with a depends, and not a `imply` or `select`.

To me changing a depends to an imply indicates to me that it will be very easy for users to create a broken configuration. At least the depends ensures that `AT_CMD` is enabled, and doesn't risk the stuck symbol behavior.

So first, this should be ensured to be correct for our users.
I can't judge whether that is achieved in current proposal, but this seems risky.

Also, if you just need access to the `AT_NOTIF` or the `AT_NOTIF_SYS_INIT` symbols for testing in `tests/` then you can provide a test only `Kconfig` that provides identical symbols that you may need without `depends`.

If you provide a bit more info, maybe there is a better solution to this.
Feel free to ping directly.

Reply:
I've done the changes according to discussion with @tejlmand. So both sms/Kconfig and at_notif/Kconfig are now restored to their original dependencies with 'select' and 'depends on' as earlier. Howeve, there is a tweak with in CONFIG_SMS_AT_CMD that handles the possibility to leave AT_CMD library out in tests to enable mocking it.